server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java

```diff
@@ -264,7 +264,7 @@ public PutMappingRequest source(Map<String, ?> mappingSource) {
         try {
             XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
             builder.map(mappingSource);
-            return source(Strings.toString(builder), XContentType.JSON);
+            return source(BytesReference.bytes(builder), builder.contentType());
```

Comment:
This change does not hurt, but isn't it ok anyways given that we create the builder ourselves and we use json as content-type?

Reply:
Right, this didn't contribute to the bug, but I just found the code easier to follow if all variants called into `#source(BytesReference source, XContentType xContentType)`.